src/NServiceBus.Testing/TestableMessageHandlerContext.cs

```diff
@@ -0,0 +1,168 @@
+
+        public IReadOnlyDictionary<string, string> MessageHeaders => new ReadOnlyDictionary<string, string>(IncomingHeaders);
+
+        public ContextBag Extensions { get; } = new ContextBag();
+
+        public SynchronizedStorageSession SynchronizedStorageSession { get; }
+
+        public bool HandlerInvocationAborted { get; private set; }
+
+        public bool HandleCurrentMessageLaterWasCalled { get; private set; }
+
+        public Task Send(object message, SendOptions options)
+        {
+            SentMessages.Add(new SentMessage<object>(message, options));
+            return Task.FromResult(0);
+        }
+
+        public Task Send<T>(Action<T> messageConstructor, SendOptions options)
+        {
+            return Send(messageCreator.CreateInstance(messageConstructor), options);
+        }
+
+        public Task Publish(object message, PublishOptions options)
+        {
+            PublishedMessages.Add(new PublishedMessage<object>(message, options));
+            return Task.FromResult(0);
+        }
+
+        public Task Publish<T>(Action<T> messageConstructor, PublishOptions publishOptions)
+        {
+            return Publish(messageCreator.CreateInstance(messageConstructor), publishOptions);
+        }
+
+        public Task Reply(object message, ReplyOptions options)
+        {
+            RepliedMessages.Add(new RepliedMessage<object>(message, options));
+            return Task.FromResult(0);
+        }
+
+        public Task Reply<T>(Action<T> messageConstructor, ReplyOptions options)
+        {
+            return Reply(messageCreator.CreateInstance(messageConstructor), options);
+        }
+
+        public Task ForwardCurrentMessageTo(string destination)
+        {
+            ForwardedMessages.Add(destination);
+            return Task.FromResult(0);
+        }
+
+        public Task HandleCurrentMessageLater()
+        {
+            HandleCurrentMessageLaterWasCalled = true;
+            return Task.FromResult(0);
+        }
+
+        public void DoNotContinueDispatchingCurrentMessageToHandlers()
+        {
+            HandlerInvocationAborted = true;
+        }
+
+        public Task Subscribe(Type eventType, SubscribeOptions options)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task Unsubscribe(Type eventType, UnsubscribeOptions options)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Validate()
+        {
+            try
+            {
+                foreach (var e in ExpectedInvocations)
+                {
+                    e.Validate(this);
+                }
+            }
+            finally
+            {
+                Clear();
+            }
+        }
+
+        public void Clear()
+        {
+            ExpectedInvocations.Clear();
+            //RepliedMessages.Clear();
+            //SentMessages.Clear();
+            //PublishedMessages.Clear();
```

Comment:
My bad, should never have been there in the first place. Its from me trying to figure stuff out